schema/omega-services.async.api.yaml

```diff
@@ -414,6 +545,230 @@ components:
+
+    AgentDescriptionSummary:
+      type: object
+      required:
+        - identifier
+        - label
+        - depository
+        - authority-file
+        - version-timestamp
+      properties:
+        identifier:
+          $ref: '#/components/schemas/GeneralIdentifier'
+        label:
+          $ref: '#/components/schemas/GeneralLabel'
+        authority-file:
+          $ref: '#/components/schemas/AuthorityFile'
+        depository:
+          $ref: '#/components/schemas/Depository'
+        version-timestamp:
+          $ref: '#/components/schemas/VersionTimestamp'
+        previous-description:
+          $ref: '#/components/schemas/GeneralIdentifier'
+
+    AgentDescriptionFull:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/AgentDescriptionSummary'
+        - $ref: '#/components/schemas/AgentDescriptionProperties'
+
+    AgentDescriptionProperties:
+      type: object
+      properties:
+        ### TODO (AR) finish this bit
+        honourific-title:
+          type: array
+          items:
+            type: string
+        title:
+          type: string
+        given-name:
+          type: array
+          items:
+            type: string
+        surname:
+          type: string
+        name:
+          type: array
+          items:
+            type: string
+        variant-name:
+          type: array
+          items:
+            type: string
+        prefererred-name:
+          type: array
+          items:
+            type: string
+        fuller-form-of-name:
+          type: string
+        member-of:
+          type: array
+          items:
+            type: object
+            required:
+              - prominent
+              - agent
+            properties:
+              prominent:
+                type: boolean
+              agent:
+                type:
+                  $ref: '#/components/schemas/GeneralIdentifier'
+        predecesor:
+          type: array
+          items:
+            type:
+              $ref: '#/components/schemas/GeneralIdentifier'
+        sucessor:
+          type: array
+          items:
+            type:
+              $ref: '#/components/schemas/GeneralIdentifier'
+        descendant:
+          type: array
+          items:
+            type:
+              $ref: '#/components/schemas/GeneralIdentifier'
+        broader-affiliation:
+          type: array
+          items:
+            type:
+              $ref: '#/components/schemas/GeneralIdentifier'
+        founder:
+          type: array
+          items:
+            type:
+              $ref: '#/components/schemas/GeneralIdentifier'
+        assoocitaed-place:
+          type: array
+          items:
+            type:
+              $ref: '#/components/schemas/GeneralIdentifier'
+        relation:
+          type: array
+          items:
+            type:
+              $ref: '#/components/schemas/GeneralIdentifier'
+        date-from:
+          type: date
```

Comment:
My reading [of this](https://www.asyncapi.com/docs/reference/specification/v2.6.0#dataTypeFormat) is that a date is of `type: string` and `format: date`. In other words you would do this:
```
date-from:
  type: string
  format: date
```

Reply:
Hmm I now see that AsyncAPI only supports JSON Schema Draft 07 - see https://www.asyncapi.com/docs/reference/specification/v3.0.0-next-major-spec.10#schemaObject

That's a shame as the latest version of JSON Schema is 2020-12 which does have specific types for `date` and `date-time`!

Reply:
I have now corrected this.